packages/@tinacms/api-git/src/router.ts

```diff
@@ -24,114 +24,55 @@ import * as express from 'express'
+  defaultCommitMessage: string
+  defaultCommitName: string
+  defaultCommitEmail: string
+  pushOnCommit: boolean
+  sshKey?: string
+  gitRemote?: string
 }
 
-/**
- *
- */
-export function checkFilePathIsInRepo(
-  filepath: string,
-  repoAbsolutePath: string
-) {
-  const fullpath = path.resolve(filepath)
-  const repopath = path.resolve(repoAbsolutePath)
-  if (fullpath.startsWith(repopath)) {
-    return true
-  } else {
-    return false
-  }
-}
-
-// Ensure remote URL is ssh
-export async function updateRemoteToSSH(pathRoot: string) {
-  const repo = await openRepo(pathRoot)
-  const remotes = await repo.getRemotes(true)
-  const originRemotes = remotes.filter((r: any) => r.name == 'origin')
-
-  if (!originRemotes.length) {
-    console.warn('No origin remote on the given repo')
-    return
-  }
-
-  const originURL = originRemotes[0].refs.push
-
-  if (originURL && !isSSHUrl(originURL)) {
-    await repo.removeRemote('origin')
-    const newRemote = getGitSSHUrl(originURL)
-    await repo.addRemote('origin', newRemote)
-  }
+const DEFAULT_OPTIONS: GitRouterConfig = {
+  pathToRepo: process.cwd(),
+  pathToContent: '',
+  defaultCommitMessage: 'Edited with TinaCMS',
+  defaultCommitName: 'TinaCMS',
+  defaultCommitEmail: 'git@tinacms.org',
+  pushOnCommit: true
 }
 
-async function createSSHKey(pathRoot: string) {
-  if (process.env.SSH_KEY) {
-    const ssh_path = path.join(pathRoot, SSH_KEY_RELATIVE_PATH)
-    const parentDir = path.dirname(ssh_path)
-    if (!fs.existsSync(parentDir)) {
-      fs.mkdirSync(parentDir, { recursive: true })
-    }
-    fs.writeFileSync(ssh_path, atob(process.env.SSH_KEY), {
-      encoding: 'utf8',
-      mode: 0o600,
-    })
-  }
+function setDefaults<T>(defaults: T, partial: Partial<T>): T {
+  return Object.assign({}, defaults, partial)
 }
 
-async function configureGitRemote(pathRoot: string) {
-  await createSSHKey(pathRoot)
-  if (process.env.GIT_REMOTE) {
-    await updateOrigin(pathRoot, process.env.GIT_REMOTE)
-  }
-  await updateRemoteToSSH(pathRoot)
-}
-
-async function updateOrigin(pathRoot: string, remote: string) {
-  const repo = await openRepo(pathRoot)
-  const newRemote = getGitSSHUrl(remote)
-
-  const existingRemotes = await repo.getRemotes(true)
-  if (existingRemotes.filter((r: any) => r.name == 'origin').length) {
-    console.warn(`Changing remote origin to ${newRemote}`)
-  }
-
-  await repo.removeRemote('origin')
-  await repo.addRemote('origin', newRemote)
-}
-
-export function router(config: GitRouterConfig = {}) {
-  const REPO_ABSOLUTE_PATH = config.pathToRepo || process.cwd()
-  const CONTENT_REL_PATH = config.pathToContent || ''
-  const CONTENT_ABSOLUTE_PATH = path.join(REPO_ABSOLUTE_PATH, CONTENT_REL_PATH)
+export function router(config: Partial<GitRouterConfig> = {}) {
+  const {
+    pathToRepo,
+    pathToContent,
+    defaultCommitMessage,
+    defaultCommitName,
+    defaultCommitEmail,
+    pushOnCommit
+  } = setDefaults(DEFAULT_OPTIONS, config)
```

Comment:
Object spread allows you to do this without directly calling Object.assign

```suggestion
  } = { ...DEFAULT_OPTIONS, ...config }
```

Reply:
Since config is a partial, will the types be correct?

Reply:
I believe so